Products/ZenModel/migrate/collectorDaemonsImportZproxy.py

```diff
@@ -0,0 +1,43 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2015, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+import Migrate
+import servicemigration as sm
+sm.require("1.0.0")
+
+
+class CollectorDaemonsImportZproxy(Migrate.Step):
+    "Import zproxy in all collector daemons."
+
+    version = Migrate.Version(5,1,0)
+
+    def cutover(self, dmd):
+        ctx = sm.ServiceContext()
+
+        # Get all services tagged "collector" and "daemon".
+        collectorDaemons = filter(lambda s: all(x in s.tags for x in ["collector", "daemon"]), ctx.services)
+
+        # Create the endpoint we're importing.
+        endpoint = sm.Endpoint(
+            name = "zproxy",
+            purpose = "import",
+            application = "zproxy",
+            portnumber = 8080,
+            protocol = "tcp",
+            addressConfig = sm.AddressConfig(0, "")
+        )
+
+        # Add the endpoint to each daemon.
+        for daemon in collectorDaemons:
+            daemon.endpoints.append(endpoint)
+
+        # Commit our changes.
+        ctx.commit()
```

Comment:
I'd say yes.  Also, keep in mind that when you run ```zenmigrate``` all migrate scripts run in one big transaction.  If any one migrate script fails, we bail on the whole thing.  FWIW we brokw this out into separate transactions for zup installs, you could do something like this if you really wanted to (https://github.com/zenoss/pm-resmgr-4.2.4/blob/master/scripts/zup/post/run_migrates)